src/views/TrackPositionAnimation/TPATimeSyncLogic.tsx

```diff
@@ -14,24 +14,23 @@ const timeComparison = (a: number, b: number) => a - b
 
 type TimeLookupFn = (time: number) => BstSearchResult<number> | undefined
 
-export const matchFocusToFrame = (animationState: AnimationState<PositionFrame>, focusTime: number | undefined, setTimeFocus: (time: number) => void) => {
-    const epsilon = 0.05
-    const currentTime = animationState?.frameData[animationState?.currentFrameIndex]?.timestamp
-    if (!currentTime || !focusTime) return
-    if (Math.abs(focusTime - currentTime) < epsilon) return
-    setTimeFocus(currentTime)
+export const matchFocusToFrame = (animationCurrentTime: number | undefined, setTimeFocus: (time: number, o: {autoScrollVisibleTimeRange?: boolean}) => void) => {
+    // const epsilon = 0.05
+    const currentTime = animationCurrentTime
+    if (currentTime === undefined) return
+    // don't let this function depend on focusTime
+    // if (Math.abs(focusTime - currentTime) < epsilon) return
+    setTimeFocus(currentTime, {autoScrollVisibleTimeRange: true})
```

Comment:
we can consider that, but it does cause that dependency problem since we'd need to know the previous value. But there may be another way around it, such as snapping to a grid -- but that can cause potential feedback problems.